pkg/occlient/occlient.go

```diff
@@ -74,8 +74,46 @@ const (
 
 	// ComponentPortAnnotationName annotation is used on the secrets that are created for each exposed port of the component
 	ComponentPortAnnotationName = "component-port"
+
+	// EnvS2IScriptsURL is an env var exposed to https://github.com/redhat-developer/odo-supervisord-image/blob/master/assemble-and-restart to indicate location of s2i scripts in this case assemble script
+	EnvS2IScriptsURL = "ODO_S2I_SCRIPTS_URL"
+
+	// EnvS2IScriptsProtocol is an env var exposed to https://github.com/redhat-developer/odo-supervisord-image/blob/master/assemble-and-restart to indicate the way to access location of s2i scripts indicated by ${${EnvS2IScriptsURL}} above
+	EnvS2IScriptsProtocol = "ODO_S2I_SCRIPTS_PROTOCOL"
+
+	// EnvS2ISrcOrBinPath is an env var exposed by s2i to indicate where the builder image expects the component source or binary to reside
+	EnvS2ISrcOrBinPath = "ODO_S2I_SRC_BIN_PATH"
+
+	// S2IScriptsURLLabel S2I script location Label name
+	// Ref: https://docs.openshift.com/enterprise/3.2/creating_images/s2i.html#build-process
+	S2IScriptsURLLabel = "io.openshift.s2i.scripts-url"
+
+	// S2ISrcOrBinLabel is the label that is provides, path where S2I expects component source or binary
+	S2ISrcOrBinLabel = "io.openshift.s2i.destination"
+
+	// EnvS2IDeploymentDir is an env var exposed to https://github.com/redhat-developer/odo-supervisord-image/blob/master/assemble-and-restart to indicate s2i deployment directory
+	EnvS2IDeploymentDir = "ODO_S2I_DEPLOYMENT_DIR"
+
+	// DefaultS2ISrcOrBinPath is the default path where S2I expects source/binary artifacts in absence of $S2ISrcOrBinLabel in builder image
+	DefaultS2ISrcOrBinPath = "/tmp/src"
```

Comment:
I believe that this should be set to `/tmp`. adding `src` should be handled on the cluster side.
Otherwise, we will have problems. As I understand it. s2i is actualy expecting source code to be in `io.openshift.s2i.destination`/src not directly in `io.openshift.s2i.destination`. That means that if image has `io.openshift.s2i.destination` set to `/my/dir` s2i expects source to be in  `/my/dir/src`.

Reply:
When we tried `/tmp` pushing source code for a python sample failed with the following logs:

```
Pushing changes to component: python-python-buxz
Please wait, building component....
+ set -eo pipefail
+ '[' -f /tmp/.s2i/bin/assemble ']'
+ '[' -n /usr/libexec/s2i ']'
+ rm -rf /opt/app-root/src/.git
+ '[' /tmp '!=' /opt/app-root/src ']'
+ cp -a /opt/app-root/src/ /tmp
+ /usr/libexec/s2i/assemble
---> Installing application source ...
mv: inter-device move failed: ‘/tmp/src/.pki’ to ‘./.pki’; unable to remove target: Directory not empty
```

Reply:
Ack